crates/relayer/src/supervisor.rs

```diff
@@ -144,6 +144,16 @@ pub fn spawn_supervisor_tasks<Chain: ChainHandle>(
         health_check(&config, &mut registry.write());
     }
 
+    // If telemetry is enabled, for each chain register the relayer's address
+    // in the list of visible fee addresses.
+    if config.telemetry.enabled {
+        for chain in registry.read().chains() {
+            if let Ok(key) = chain.get_key() {
+                telemetry!(add_visible_fee_address, key.account());
+            }
```

Comment:
We need to prefix variables only used in the telemetry! block with `_` to avoid unused variable warnings when built without the telemetry feature.

```suggestion
            if let Ok(_key) = chain.get_key() {
                telemetry!(add_visible_fee_address, _key.account());
            }
```